projects/core/src/occ/services/occ-endpoints.service.ts

```diff
@@ -58,52 +86,138 @@ export class OccEndpointsService {
    * @param endpoint Endpoint suffix
    */
   getOccEndpoint(endpoint: string): string {
-    if (!this.config?.backend?.occ) {
-      return '';
-    }
-    endpoint = this.config.backend.occ.endpoints?.[endpoint];
-
-    if (
-      !endpoint.startsWith('/') &&
-      !this.config.backend.occ.prefix.endsWith('/')
-    ) {
-      endpoint = '/' + endpoint;
-    }
+    endpoint = this.getRawEndpointValue(endpoint);
 
-    return (
-      this.config.backend.occ.baseUrl +
-      this.config.backend.occ.prefix +
-      endpoint
-    );
+    return this.getEndpoint(endpoint, { baseSite: false });
   }
 
   /**
-   * Returns base OCC endpoint (baseUrl + prefix + baseSite)
+   * @Deprecated since 3.2 - use "buildOccBaseUrl" with the same parameters
+   *
+   * Returns base OCC endpoint (baseUrl + prefix + baseSite) by if no parameters are specified
+   *
+   * @param propertiesToOmit Specify properties to not add to the url (baseUrl, prefix, baseSite)
    */
-  getBaseEndpoint(): string {
+  getBaseEndpoint(propertiesToOmit?: BaseOccUrlProperties): string {
     if (!this.config?.backend?.occ) {
       return '';
     }
 
-    return (
-      (this.config.backend.occ.baseUrl || '') +
-      this.config.backend.occ.prefix +
-      this.activeBaseSite
-    );
+    return this.getBaseUrl(propertiesToOmit);
   }
 
   /**
+   * @Deprecated since 3.2 - use "buildUrlFromEndpointString" with the same parameters
+   *
    * Returns an OCC endpoint including baseUrl and baseSite
+   *
    * @param endpoint Endpoint suffix
+   * @param propertiesToOmit Specify properties to not add to the url (baseUrl, prefix, baseSite)
    */
-  getEndpoint(endpoint: string): string {
-    if (!endpoint.startsWith('/')) {
+  getEndpoint(
+    endpoint: string,
+    propertiesToOmit?: BaseOccUrlProperties
+  ): string {
+    if (!endpoint.startsWith('/') && !this.getPrefix().endsWith('/')) {
       endpoint = '/' + endpoint;
     }
-    return this.getBaseEndpoint() + endpoint;
+    return this.buildUrlFromEndpointString(endpoint, propertiesToOmit);
+  }
+
+  /**
+   * Returns base OCC endpoint (baseUrl + prefix + baseSite) base on provided values
+   *
+   * @param baseUrlProperties Specify properties to not add to the url (baseUrl, prefix, baseSite)
+   */
+  getBaseUrl(
+    baseUrlProperties: BaseOccUrlProperties = {
+      baseUrl: true,
+      prefix: true,
+      baseSite: true,
+    }
+  ): string {
+    const baseUrl =
+      baseUrlProperties.baseUrl === false
+        ? ''
+        : this.config.backend.occ.baseUrl;
+    const prefix = baseUrlProperties.prefix === false ? '' : this.getPrefix();
+    const baseSite =
+      baseUrlProperties.baseSite === false ? '' : this.activeBaseSite;
+
+    return urlPathJoin(baseUrl, prefix, baseSite);
```

Comment:
I think it will fail in one case. When we want only baseSite url it won't start with `/` and I would assume that this method either returns url starting with `/` or starting with `http`.

Reply:
With the current implementation it won't add a `/` to the beginning of your url, it will simply join the parts. That said if you give it only a baseSite with this value `/base-site` it will return it as is but if you don't pass a value with a `/`, it won't add it.

To be honest I don't  really see the value in using this method if you only want to get the base site. If we want to start adding `/` at the start for certain conditions it will add a bit of complexity to the method.

Reply:
Ok. Let's leave it as it is. If someone reports this use case in the future we will fix it.